### PR TITLE
Version bump + add max_input_length to gptq 

### PR DESCRIPTION
**Pull request by SunMarc**
# What does this PR do ? 
This PR bumps the minimum required version of `auto_gptq` and adds the possibility to change the max input length when using exllama + act_order with gptq. 
We also do a couple of fixes: 
- disable exllama kernel when doing quantization with act order because the user may save the model with the weights in the wrong order. exllama + act_order should only be used for inference and the user should not save the model. This is a temporary measure until we are able to reorder the weights.
- `self.quant_method = QuantizationMethod.GPTQ `for people who wants to quantize with optimum api and load it with transformers like this [tutorial](https://www.philschmid.de/gptq-llama) .

### CODE REVIEW

File: tests/gptq/test_quantization.py [hunk 1]
```diff
@@ -126,14 +128,89 @@ def test_serialization(self):
...
+            quantized_model_from_saved = load_quantized_model(
+                empty_model, save_folder=tmpdirname, device_map={"": 0}, disable_exllama=False, max_input_length=4028
+            )
+
+            prompt = "I am in Paris and" * 1000
+            inp = self.tokenizer(prompt, return_tensors="pt").to(0)
+            self.assertTrue(inp["input_ids"].shape[1] > 4028)
+            with self.assertRaises(RuntimeError) as cm:
+                quantized_model_from_saved.generate(**inp, num_beams=1, min_new_tokens=3, max_new_tokens=3)
+                self.assertTrue("temp_state buffer is too small" in str(cm.exception))
```
Comment (fxmarty) [inline tests/gptq/test_quantization.py:208]: The error shown from AutoGPTQ is irrelevant if the model is loaded through transformers right? Should I do a fix on AutoGPTQ side to have the exllama_set_max_input_length method behave well with transformers?

**Review by fxmarty — APPROVED** [review]
LGTM thank you!

**Review by younesbelkada — APPROVED** [review]
Thanks a lot @SunMarc !

File: tests/gptq/test_quantization.py (hunk 1 above)
Comment (fxmarty) [reply tests/gptq/test_quantization.py:208]: Calling `exllama_set_max_input_length` directly on a transformers model will fail, see: https://github.com/PanQiWei/AutoGPTQ/issues/315

Comment (fxmarty) [reply tests/gptq/test_quantization.py:208]: Why is this fine? The error shown by AutoGPTQ suggests to use `exllama_set_max_input_length` which does not work with transformers.

Comment (SunMarc) [reply tests/gptq/test_quantization.py:208]: What is the fix that you want to do ? I think i'm missing something 😅

Comment (SunMarc) [reply tests/gptq/test_quantization.py:208]: That would be great indeed ! The workaround that is used for now in optimum/transformers is: 
```py 
class StoreAttr(object):
    pass

model.quantize_config = StoreAttr()
model.quantize_config.desc_act = self.desc_act
```

**Pull request merged**